### ISSUE TITLE
Bug: [@typescript-eslint/no-explicit-any] in catch (ex: any) can't work for both normal ESLint and TS ESLint

### ISSUE THREAD

**Issue by ajvincent** [typescript-eslint/typescript-eslint#5322]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.2&sourceType=module&code=DYUwLgBADghgznCBeCAzGw4gNwCgwBOAnhAN64QRgAWBA9gO4QB2ITAogfQQBQBEqOnT4BKPAF9cAeikQQcYAEtmYALQATRXBgAjUKtYAPNUtYQAAmCJR5AYwKKoa+abBTmdVSENQltxWowzES4tjBgttQQPN4AXBBBRCJkFBCKqNHeAHQAtvLaAOYgyEgoAkKiqZSwCMhUBACuOLiSQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge3oA9jGzKdAIZNC6KImjQO0SODABfEAqA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
let pass = false;
try {
  throw new Error("foo");
}
// eslint-disable-next-line @typescript-eslint/no-explicit-any
catch (ex: any) {
  if (ex.message === "foo")
    pass = true;
}
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/no-explicit-any": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

`@typescript-eslint/no-explicit-any` should not apply to catch exception variables.

### Actual Result

- If I feed this to standard eslint, I get an exception for `Definition for rule '@typescript-eslint/no-explicit-any' was not found  @typescript-eslint/no-explicit-any`.  (This is expected, but annoying.)
- If I remove the `eslint-disable-next-line`, I trigger the rule.
- If I change the catch line to `catch (ex: Error) {`, I trigger a TypeScript exception: `Catch clause variable type annotation must be 'any' or 'unknown' if specified.ts(1196)`
- If I change the catch line to `catch (ex: unknown) {`, TypeScript disallows checking properties of the exception.
- If I disable `@typescript-eslint/no-explicit-any` globally, I lose that protection across many files.

In short, I cannot have a try...catch statement with a typed exception, run both against standard eslint and TypeScript's eslint plugin.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.30.5` |
| `@typescript-eslint/parser`            | `5.30.5` |
| `TypeScript`                           | `4.7.4` |
| `ESLint`                               | `8.19.0` |
| `node`                                 | `16.14.0` |

**Comment by ajvincent** [comment]
At least there's a workaround I didn't think of initially:
```typescript
let pass = false;
try {
  throw new Error("foo");
}
catch (ex: unknown) {
  if ((ex as Error).message === "foo")
    pass = true;
}
```

**Comment by bradzacher** [comment]
You don't need an annotation on the catch variable. It defaults to `any` unless you use [`useUnknownInCatchVariables`] (https://www.typescriptlang.org/tsconfig#useUnknownInCatchVariables).

If you're explicitly annotating the variable with `any` then I see no reason that the rule shouldn't report on it.
